tests/headers/clte_test.go

```diff
@@ -355,3 +300,229 @@ func TestTransferEncodingChunkSizeLength(t *testing.T) {
+			}
+		})
+		if err != nil {
+			t.Fatalf("MakeRequest() got err: %v want: nil", err)
+		}
+
+		if got, want := extractStatus(resp), statusOK; got != want {
+			t.Errorf("status code got: %q want: %q", got, want)
+		}
+	})
+
+	t.Run("Desired behavior", func(t *testing.T) {
+		t.Skip()
+		resp, err := requesttesting.MakeRequest(context.Background(), request, func(r *http.Request) {
+			t.Error("Expected handler to not be called!")
+		})
+		if err != nil {
+			t.Fatalf("MakeRequest() got err: %v want: nil", err)
+		}
+
+		if got, want := extractStatus(resp), statusBadRequest; got != want {
+			t.Errorf("status code got: %q want: %q", got, want)
+		}
+	})
+}
+
+func TestTransferEncodingIdentity(t *testing.T) {
+	// net/http currently accepts the identity transfer encoding and
+	// interprets it as if there was no transfer encoding at all.
+	//
+	// But RFC 7230 has deprecated and removed the identity encoding.
+	// The net/http package should therefore remove support for this
+	// encoding. The Desired behavior is therefore to return a
+	// 501 (Not Implemented).
+
+	request := []byte("GET / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Length: 11\r\n" +
+		"Transfer-Encoding: identity\r\n" +
+		"\r\n" +
+		"5\r\n" +
+		"ABCDE\r\n" +
+		"0\r\n" +
+		"\r\n")
+
+	t.Run("Current behavior", func(t *testing.T) {
+		resp, err := requesttesting.MakeRequest(context.Background(), request, func(r *http.Request) {
+			wantHeaders := map[string][]string{"Content-Length": []string{"11"}}
+			if diff := cmp.Diff(wantHeaders, map[string][]string(r.Header)); diff != "" {
+				t.Errorf("r.Header mismatch (-want +got):\n%s", diff)
+			}
+
+			if diff := cmp.Diff([]string(nil), r.TransferEncoding); diff != "" {
+				t.Errorf("r.TransferEncoding mismatch (-want +got):\n%s", diff)
+			}
+
+			if want := int64(11); r.ContentLength != want {
+				t.Errorf("r.ContentLength got: %v want: %v", r.ContentLength, want)
+			}
+
+			body, err := ioutil.ReadAll(r.Body)
+			if err != nil {
+				t.Fatalf("ioutil.ReadAll(r.Body) got err: %v", err)
+			}
+
+			if got, want := string(body), "5\r\nABCDE\r\n0"; got != want {
+				t.Errorf("r.Body got: %q want: %q", got, want)
+			}
+		})
+		if err != nil {
+			t.Fatalf("MakeRequest() got err: %v", err)
+		}
+
+		if got, want := extractStatus(resp), statusOK; got != want {
+			t.Errorf("status code got: %q want: %q", got, want)
+		}
+	})
+
+	t.Run("Desired behavior", func(t *testing.T) {
+		t.Skip()
+		resp, err := requesttesting.MakeRequest(context.Background(), request, func(r *http.Request) {
+			t.Error("Expected handler to not be called!")
+		})
+		if err != nil {
+			t.Fatalf("MakeRequest() got err: %v", err)
+		}
+
+		if got, want := extractStatus(resp), statusNotImplemented; got != want {
+			t.Errorf("status code got: %q want: %q", got, want)
+		}
+	})
+}
+
+func TestTransferEncodingListIdentityFirst(t *testing.T) {
+	// In TestTransferEncodingIdentity we see that the
```

Comment:
I'd avoid pointing the user to the previous test and maybe try to (re)explain the entire issue here.